pkg/controllers/common/conditions.go

```diff
@@ -2,13 +2,67 @@ package common
 
 import (
 	"strings"
+	"time"
 
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	operatorv1 "github.com/openshift/api/operator/v1"
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
+var _ error = &ControllerProgressingError{}
+
+type ControllerProgressingError struct {
```

Comment:
mini nit: i guess this could be package private (lower case)

Reply:
I think errors should better be exported in case someone wants to check on a kind of an error, even though such use might be unlikely in this case